Added more known type-hints

Quix0r opened the pull request:
This PR adds some more type-hints to method signatures.

Quix0r committed:
Added more known type-hints

Signed-off-by: Roland Häder <roland@mxchange.org>

annando commented:
We are really short before creating the release. In this stage I think it would be best to only have bugfix PRs. type hints are surely a good thing, but they can cause side effects that may not be obvious at the first glance. I'm not sure what the others are thinking about this, but possibly it would be a good idea to change this PR so that it is against the develop branch.

nupplaphil commented:
I'm with @annando - We already see an error in the PHP Lint testcase, so we shouldn't push such changes at this late stage

[edit] but @Quix0r don't get me wrong -->  I really, really like the idea of type-hinting :)

MrPetovan commented:
I've moved the base branch to `develop` so that this PR doesn't get automatically closed when we delete the RC branch after release.

MrPetovan commented:
Rebased against the latest `develop`, we can review now.

MrPetovan reviewed (changes requested):
From PHP linter
> Fatal error: A function with return type must return a value in ./src/Model/Item.php on line 2814

Quix0r committed:
Added more type-hints

Signed-off-by: Roland Häder <roland@mxchange.org>

Quix0r commented:
> From PHP linter
> 
> > Fatal error: A function with return type must return a value in ./src/Model/Item.php on line 2814

I have there an empty line and https://github.com/friendica/friendica/pull/11624/files doesn't show any changes there.

Quix0r commented:
Also remember that there is a difference between `develop` and `2205/6-rc` for the method `prepareBody()` as 2205/6-rc adds another parameter to the method declaration.

Quix0r commented:
Corrected my rebase on updated `develop` changes.

MrPetovan commented:
> > From PHP linter
> > > Fatal error: A function with return type must return a value in ./src/Model/Item.php on line 2814
> 
> I have there an empty line and https://github.com/friendica/friendica/pull/11624/files doesn't show any changes there.

You can't have `return;` in a function that has an explicit `: string` return type hint. You need to return en empty string at least.

MrPetovan commented on src/Model/Item.php:
```diff
@@ -2769,7 +2769,7 @@ private static function putInCache(&$item)
 	 * @hook  prepare_body_final ('item'=>item array, 'html'=>body string) after attach icons and blockquote special case handling (spoiler, author)
 	 */
-	public static function prepareBody(array &$item, $attach = false, $is_preview = false, $only_cache = false)
+	public static function prepareBody(array &$item, bool $attach = false, bool $is_preview = false, bool $only_cache = false): string
```
You still need to make sure all the `return` calls of this method are returning a string as type-hinted.

Quix0r replied:
I followed strictly the documentation, `void` shouldn't be together with any other type (`string` here). So can I change `return;` in this method to `return '';` then? I know what you do here, simply get out of the method as the below code should be only executed when `$only_cache` is `false`.

MrPetovan replied:
Yes, you can (and you should change it). It doesn't make sense for a string processing method to return void.

Quix0r replied:
Now I see it on my server, too: `PHP Fatal error:  A function with return type must return a value in /var/www/.../src/Model/Item.php on line 2814`.

Quix0r replied:
Okay, I changed it.

Quix0r committed:
Changes:
- added more type-hints
- changed `return;` to `return '';` when `string` was requested as returned type-hint
  (thanks to @MrPetovan)

Quix0r committed:
Changes:
- ops, wasn't actually fixing `return;` to proper `return '';`
- added more type-hints
- added TODO ($uid is unused)

tobiasd commented:
This PR seems to be causing https://github.com/friendica/friendica/issues/11630#issuecomment-1157401304 (I've reverted my node to the PR before and the error is gone). Might be also the reason why @Quix0r cannot follow Mastodon accounts? (see the [forum posting](https://forum.friendi.ca/display/adbba6c6-1862-aae4-04e2-38f538779582))

Quix0r commented:
I have fixed the 2 wrong calls, I guess `$message` in `HTML::toBBCode()` should remain `string` and we need to locate all bad calls.
